clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1721,6 +1721,40 @@ static void CollectARMPACBTIOptions(const ToolChain &TC, const ArgList &Args,
   }
 }
 
+static void EmitAccuracyDiag(const Driver &D, const JobAction &JA,
+                             StringRef AccuracValStr, StringRef TargetPrecStr) {
+  if (JA.isDeviceOffloading(Action::OFK_SYCL)) {
+    D.Diag(clang::diag::warn_acuracy_conflicts_with_explicit_target_prec_option)
+        << AccuracValStr << TargetPrecStr;
+  }
+}
+
+void Clang::AddSPIRTargetArgs(const ArgList &Args, ArgStringList &CmdArgs,
+                              const JobAction &JA, const Driver &D) const {
+  if (JA.isDeviceOffloading(Action::OFK_SYCL)) {
+    if (Arg *A = Args.getLastArg(options::OPT_ffp_model_EQ)) {
+      if (!strcmp(A->getValue(), "fast")) {
+        CmdArgs.push_back("-fno-offload-fp32-prec-div");
+        CmdArgs.push_back("-fno-offload-fp32-prec-sqrt");
+      }
```

Comment:
Should we allow users to override with `-foffload-fp32-prec-div|sqrt`?
```suggestion
      if (!strcmp(A->getValue(), "fast")) {
        if (!Args.hasFlag(option::OPT_foffload_fp32_prec_div,
                          option::OPT_fno_offload_fp32_prec_div, false))
          CmdArgs.push_back("-fno-offload-fp32-prec-div");
        if (!Args.hasFlag(option::OPT_foffload_fp32_prec_sqrt,
                          option::OPT_fno_offload_fp32_prec_sqrt, false))
          CmdArgs.push_back("-fno-offload-fp32-prec-sqrt");
      }
```

Reply:
Not sure. I would think that users could choose to compile with:
`clang -fsycl -ffp-model=fast -foffload-fp32-prec-sqrt hello.cpp`
or:
`clang -fsycl -foffload-fp32-prec-sqrt -ffp-model=fast hello.cpp`
These shouldn't give the same result. In the first one, the sqrt results are precise. In the second one, they are rounded.

I think that's just following the last command wins rule. In which case we need a compilated process here to find the order in which the options interact with one another.

Reply:
Hmm... If that's the case we may want to integrate the logic into where all of the other FP model options are being manipulated (the larger for loop here: https://github.com/intel/llvm/blob/bdf78d7d165f9022397f885590c4889eb6513682/clang/lib/Driver/ToolChains/Clang.cpp#L2994) and only add the `-cc1` option under the `IsDeviceOffloading` condition.

Reply:
Okay and that would work for OpenMP too!